index.js

```diff
@@ -50,7 +52,8 @@ module.exports = function(parent, theirs, mine, options = {}) {
  * @param {object} options - Key specific options for diff
  * @returns {Array} - Array of objects specifying the differences
  */
-function recurse(parent, theirs, mine, path = [], options = {}) {
+function recurse(parent, theirs, mine, path, options) {
+  path = path || [];
```

Comment:
As discussed out loud, protect against the function erroring out depending on parameters.